dolphin/db/sqlalchemy/api.py

```diff
@@ -85,15 +92,84 @@ def register_db():
         model.metadata.create_all(engine)
 
 
+def _process_model_like_filter(model, query, filters):
+    """Applies regex expression filtering to a query.
+
+    :param model: model to apply filters to
+    :param query: query to apply filters to
+    :param filters: dictionary of filters with regex values
+    :returns: the updated query.
+    """
+    if query is None:
+        return query
+
+    for key in sorted(filters):
+        column_attr = getattr(model, key)
+        if 'property' == type(column_attr).__name__:
+            continue
+        value = filters[key]
+        if not (isinstance(value, (six.string_types, int))):
+            continue
+        query = query.filter(
+            column_attr.op('LIKE')(u'%%%s%%' % value))
+    return query
+
+
+def apply_like_filters(model):
+    def decorator_filters(process_exact_filters):
+        def _decorator(query, filters):
+            exact_filters = filters.copy()
+            regex_filters = {}
+            for key, value in filters.items():
+                # NOTE(tommylikehu): For inexact match, the filter keys
+                # are in the format of 'key~=value'
+                if key.endswith('~'):
+                    exact_filters.pop(key)
+                    regex_filters[key.rstrip('~')] = value
+            query = process_exact_filters(query, exact_filters)
+            return _process_model_like_filter(model, query, regex_filters)
+        return _decorator
+    return decorator_filters
+
+
+def is_valid_model_filters(model, filters, exclude_list=None):
+    """Return True if filter values exist on the model
+
+    :param model: a Dolphin model
+    :param filters: dictionary of filters
+    """
+    for key in filters.keys():
+        if exclude_list and key in exclude_list:
+            continue
+        if key == 'metadata':
+            if not isinstance(filters[key], dict):
+                LOG.debug("Metadata filter value is not valid dictionary")
+                return False
+            continue
+        try:
+            key = key.rstrip('~')
+            getattr(model, key)
+        except AttributeError:
+            LOG.debug("'%s' filter key is not valid.", key)
+            return False
+    return True
+
+
 def access_info_create(context, values):
     """Create a storage access information."""
-    register_ref = models.AccessInfo()
-    this_session = get_session()
-    this_session.begin()
-    register_ref.update(values)
-    this_session.add(register_ref)
-    this_session.commit()
-    return register_ref
+    if not values.get('storage_id'):
+        values['storage_id'] = uuidutils.generate_uuid()
```

Comment:
storage_id should not be return by driver, it's a database id. for each creation operaion, uuid will be created if clients (such as api, driver manager, task manager) don't specify it in the model.

Reply:
we discussed to move this to driver manager_ , if storage_id is not provided by driver, driver manager generates a uuid..

Reply:
I didn't get your point here, it's a db implementation, the logic flow is if it doesn't contain id, it will generate, should not depend on the driver manager here. Uuid generation is also handled by driver manager

Reply:
fine.. hope it is ok to have double checking here..

Reply:
IMHO, this is ok to have double check but not a good practice. Anything related to storage attrib/ID should be handled till driver/driver manager layer and should not be exposed above. If the client doesn't get storage_id, it should throw exception and not try to create one.

Reply:
no double check here, the code help caller to generate the uuid if the caller doesn't generate.

Reply:
Ok..